spec/views/posts/posts_index_page_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'rails_helper'
+
+user = FactoryBot.create(:user)
+post = FactoryBot.create(:post_with_comments, author: user)
+
+RSpec.describe 'post index page', type: :feature do
+  scenario 'display user profile picture' do
+    visit user_posts_path(post.author)
+    expect(page).to have_css("img[src='#{post.author.photo}']")
+  end
+
+  scenario 'display username' do
+    visit user_posts_path(post.author)
+    expect(page).to have_content(post.author.name)
+  end
+
+  scenario 'display number of posts' do
+    visit user_posts_path(post.author)
+    expect(page).to have_content(post.author.posts_counter)
+  end
+
+  scenario 'display post title' do
+    visit user_posts_path(post.author)
+    expect(page).to have_content(post.title)
+  end
+
+  scenario 'display post text' do
+    visit user_posts_path(post.author)
+    expect(page).to have_content(post.text)
+  end
+
+  scenario 'display number of comments' do
+    visit user_posts_path(post.author)
+    expect(page).to have_content(post.comments_counter)
+  end
+
+  scenario 'display number of likes' do
+    visit user_posts_path(post.author)
+    expect(page).to have_content(post.likes_counter)
+  end
+
+  scenario 'display first comment' do
+    comment = post.comments.first
+
+    visit user_posts_path(post.author)
+    expect(page).to have_content(comment.text)
+  end
+
+  scenario 'when clicked redirects to post show page' do
+    visit user_posts_path(post.author)
+    click_link(href: user_post_path(post.author, post))
+    expect(page).to have_content(post.author.name)
+  end
```

Comment:
- [ ] @team great work adding a test for the posts page, however, you forgot to add tests for the `see a section for pagination if there are more posts than fit on the view`, this is part of the tests required as shown below.
![image](https://github.com/shasherazi/blog/assets/61408860/163c9849-71b3-435a-aa76-27e135b133e6)